plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -195,6 +198,35 @@ func (c *nodePlugin) admitPodStatus(nodeName string, a admission.Attributes) err
 	}
 }
 
+func (c *nodePlugin) admitPodEviction(nodeName string, a admission.Attributes) error {
+	switch a.GetOperation() {
+	case admission.Create:
+		// require eviction to an existing pod object
+		eviction, ok := a.GetObject().(*policy.Eviction)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		// get the existing pod from the server cache
+		existingPod, err := c.podsGetter.Pods(eviction.Namespace).Get(eviction.Name, v1.GetOptions{ResourceVersion: "0"})
```

Comment:
I would expect the namespace to be pulled from the admission attributes, and the pod name to be pulled from the admission attributes as well, if set, rather than from the object